forge-game/src/main/java/forge/game/zone/MagicStack.java

```diff
@@ -789,18 +800,23 @@ public boolean addAllTriggeredAbilitiesToStack() {
             playerTurn = game.getNextPlayerAfter(playerTurn);
         }
 
-        Player whoAddsToStack = playerTurn;
-        do {
-            result |= chooseOrderOfSimultaneousStackEntry(whoAddsToStack, false);
-            // 2014-08-10 Fix infinite loop when a player dies during a multiplayer game during their turn
-            whoAddsToStack = game.getNextPlayerAfter(whoAddsToStack);
-        } while (whoAddsToStack != null && whoAddsToStack != playerTurn);
-        // 603.3b (Strict Proctor)
-        whoAddsToStack = playerTurn;
-        do {
-            result |= chooseOrderOfSimultaneousStackEntry(whoAddsToStack, true);
-            whoAddsToStack = game.getNextPlayerAfter(whoAddsToStack);
-        } while (whoAddsToStack != null && whoAddsToStack != playerTurn);
+        // Grab players in turn order starting with the active player
+        List<Player> players = game.getPlayersInTurnOrder(playerTurn);
```

Comment:
Looks like the logic there is wrong?
you need to shift them in the other direction instead

Reply:
Why do you think the logic is wrong? We start at the current player, then iterate through?

Reply:
see Javadoc for Collections.rotate:
> For example, suppose list comprises [t, a, n, k, s].After invoking Collections.rotate(list, 1) (or Collections.rotate(list, -4)), list will comprise [s, t, a, n, k]. 

you aren't ensuring APNAP

Reply:
Ah I see.. the rotate isn't doing what i expected it to do. I'll run some tests on it to make sure its rotating as expected.